Fix locale-dependent float printing in GGUF metadata

ssam18 opened the pull request:
I was running some llama.cpp examples on a system with a German locale (de_DE) and noticed something odd - when llama-cli printed out the model metadata, all the float values had commas as decimal separators (like "0,000000") instead of periods. But when I ran llama-perplexity on the same model, it used periods normally.

After some digging, I found the issue was in the gguf_data_to_str() function in llama-impl.cpp. It was using std::to_string() to format floats, which respects the system's LC_NUMERIC locale setting. So depending on which tool you used and what locale it was running with, you'd get different formatting.

I've changed it to use std::ostringstream with std::locale::classic() instead, which always formats floats with a period as the decimal separator, regardless of the system locale. This should make the output consistent across all tools and locales.

Fixes #10613

ggerganov commented:
@JohannesGaessler Can you confirm it fixes https://github.com/ggml-org/llama.cpp/issues/10613?

ssam18 committed:
Set C locale for consistent float formatting across all binaries.

JohannesGaessler commented:
Originally all binaries were in the `examples` folder but nowadays there are also binaries in the `tools` folder (which I assume you overlooked?).

ssam18 committed:
Add C locale setting to all tools binaries

Add std::setlocale(LC_NUMERIC, "C") to all 16 binaries in the tools/
directory to ensure consistent floating-point formatting.

SamareshSingh commented on tools/llama-bench/llama-bench.cpp:
```diff
@@ -2000,6 +2000,7 @@ static std::unique_ptr<printer> create_printer(output_formats format) {
 int main(int argc, char ** argv) {
+    std::setlocale(LC_NUMERIC, "C");
     // try to set locale for unicode characters in markdown
     setlocale(LC_CTYPE, ".UTF-8");
```
yes, that is true. using std namespace gives clarity on the function.

ngxson commented:
@JohannesGaessler for all examples don't use it, we can set it explicitly on main.

I think doing it in `common_params_parse` does make sense though, as the libcommon also have various code paths to setup the CLI environment, including logging system, console, etc. So adding this as a "formatting setup" does make sense IMO

JohannesGaessler reviewed (changes requested):
I don't see us changing global settings anywhere in `common_params_parse`. That is fundamentally not something that a library should be doing upon use unless it is done by an explicit and dedicated function.

JohannesGaessler committed:
Apply suggestion from @JohannesGaessler

JohannesGaessler reviewed (approved):
Yes, the CI failures seem to be unrelated.
